src/utils/number.ts

```diff
@@ -26,20 +26,25 @@ export function parseNumberString(numberString?: string): string {
 }
 
 export function sanitizeDecimalString(decimalString: string): string {
-  return decimalString?.endsWith(".") ? decimalString.replace(".", "") : decimalString;
+  return decimalString.replace(/\.$/, "");
 }
 
 export function sanitizeNegativeString(negativeString: string): string {
-  return negativeString.charAt(0) + negativeString.substring(1).replace(/-/g, "");
+  return negativeString.replace(/(?!^-)-/g, "");
 }
 
 export function sanitizeLeadingZeroString(zeroString: string): string {
-  return zeroString.replace(/^0+$/, "0");
+  return zeroString.replace(/(?=^[-0])^([-0])0+(?=\d)/, `$1`);
```

Comment:
Does the updated version also sanitize the non-negative case? Wondering if it needs to be renamed. Also, could the pattern be stored in a local var with an expressive name to convey what it will capture?

Reply:
It does capture non-zero as well. `[-0]` is hyphen or zero. I will move all of these to a local var for maintainability

Reply:
Thanks for clarifying. I always need a double take on RegExps.

![confused-marky](https://user-images.githubusercontent.com/197440/147595412-8e95118b-8fe8-4872-a5f9-95093fcd3335.gif)